### ISSUE TITLE
CMake argument documentation in Sphinx Doc

### ISSUE THREAD

**Issue by SimeonEhrig** [alpaka-group/alpaka#1045]
I think a documentation about the different CMake arguments in the Sphinx documentation would be really helpful. For example, today I had some difficulties compiling my Alpaka code with the HIP backend on an AMD system because I didn't know that I had to set the `ALPAKA_HIP_PLATFORM` cmake argument to `clang`. Another really common workflow for me is to search for the backend names in the cmake files, that I can disable it in my project.

@alpaka-group/alpaka-developers 
Do you also think that a CMake documentation is necessary?

**Comment by ax3l** [comment]
Definitely, maybe add a table including re-iteration of standard but very common CMake options that a new developer that is new to CMake, too does not know:

Examples:
- https://openpmd-api.readthedocs.io/en/0.11.1-alpha/dev/buildoptions.html
- https://warpx.readthedocs.io/en/latest/building/cmake.html

**Comment by j-stephan** [comment]
Yes, that is a good idea!

**Comment by SimeonEhrig** [comment]
CMake options documentation is planned for the restructuring of the Sphinx Doc, which is work in progress (I hope, I can create a draft this week). The new structure should be similar to the [pybind11 doc](https://pybind11.readthedocs.io/en/stable/):

* Basic
* Advanced
* Extra Information
* Development/Contribution

At the moment, I plan to add a site for each accelerator to the Advanced section. So I think, it is a good idea to distribute the CMake arguments to a general CMake args site and the accelerator depend arguments to the accelerator site.
